src/script/lua_api/l_object.cpp

```diff
@@ -391,35 +410,36 @@ int ObjectRef::l_set_physics_override(lua_State *L)
 	PlayerSAO *co = (PlayerSAO *) getobject(ref);
 	if (co == NULL) return 0;
 	// Do it
-	if (lua_istable(L, 2)) {
-		co->m_physics_override_speed = getfloatfield_default(
-				L, 2, "speed", co->m_physics_override_speed);
-		co->m_physics_override_jump = getfloatfield_default(
-				L, 2, "jump", co->m_physics_override_jump);
-		co->m_physics_override_gravity = getfloatfield_default(
-				L, 2, "gravity", co->m_physics_override_gravity);
-		co->m_physics_override_sneak = getboolfield_default(
-				L, 2, "sneak", co->m_physics_override_sneak);
-		co->m_physics_override_sneak_glitch = getboolfield_default(
-				L, 2, "sneak_glitch", co->m_physics_override_sneak_glitch);
+	if (read_physics_modifier(L, 2, co->m_physics_override)) {
 		co->m_physics_override_new_move = getboolfield_default(
 				L, 2, "new_move", co->m_physics_override_new_move);
 		co->m_physics_override_sent = false;
+		co->m_physics_override_set = true;
 	} else {
 		// old, non-table format
 		if (!lua_isnil(L, 2)) {
-			co->m_physics_override_speed = lua_tonumber(L, 2);
+			co->m_physics_override.speed = lua_tonumber(L, 2);
 			co->m_physics_override_sent = false;
+			co->m_physics_override_set = true;
 		}
 		if (!lua_isnil(L, 3)) {
-			co->m_physics_override_jump = lua_tonumber(L, 3);
+			co->m_physics_override.jump = lua_tonumber(L, 3);
 			co->m_physics_override_sent = false;
+			co->m_physics_override_set = true;
 		}
 		if (!lua_isnil(L, 4)) {
-			co->m_physics_override_gravity = lua_tonumber(L, 4);
+			co->m_physics_override.gravity = lua_tonumber(L, 4);
 			co->m_physics_override_sent = false;
+			co->m_physics_override_set = true;
 		}
 	}
+
+		if (!co->getPhysicsModifiers().empty()) {
```

Comment:
Misplaced tabulator. Also flush after the script backtrace.
Suggestion for above: Maybe reset `co->m_physics_override_set` to `false` so that the physics modifiers can be applied in case all physics override params are back to their defaults.